MyBlog.Services/Areas/Users/UserService.cs

```diff
@@ -57,15 +55,16 @@ public async Task<int> CreateAsync(UserDtoInput userInput)
         await _context.Users.AddAsync(user);
         await _context.SaveChangesAsync();
         
-        return user.Id;
+        var userDto = _mapper.Map<UserDto>(user);
+        
+        return userDto;
     } 
 
-    public async Task<int> UpdateByIdAsync(int id, UserDtoInput userInput)
+    public async Task<UserDto> UpdateByIdAsync(int id, UserDtoInput userInput)
     {
-        var user = await _context.Users.FirstOrDefaultAsync(u => u.Id == id)
-                   ?? throw new NotFoundException($"User with Id: {id} is not found");
+        var user = await GetUserByIdAsync(id);
 
-        if (CheckForFreeAsync(id, userInput).Result == false)
+        if (await CheckEmailAndPhoneForFreeAsync(id, userInput) == false)
         {
```

Comment:
you are in the if case, so if you use this bool expression, false condition replace to the "!"